test_simple_graph.py

```diff
@@ -41,9 +69,22 @@ def test_add_edge(populated):
     assert [7, 5] in populated.edges()
 
 
+def test_add_edge_new_node(populated):
+    populated.add_edge(5, 28)
```

Comment:
I'm a firm believer that every test needs a docstring. Most of your test names are pretty self-evident, so you don't lose a lot by not having docstrings, but this function absolutely requires some added explanation. I had no idea what was going on until I scrolled up to your add_edge function and realized that adding an edge also can add a node.